packages/react-components/react-accordion/src/components/AccordionHeader/useAccordionHeader.tsx

```diff
@@ -55,19 +50,23 @@ export const useAccordionHeader_unstable = (
       expandIcon: 'span',
       icon: 'div',
     },
-    root: getNativeElementProps(as || 'div', {
-      ref,
-      ...props,
-    }),
-    icon: resolveShorthand(icon),
-    expandIcon: resolveShorthand(expandIcon, {
-      required: true,
+    root: slot.always(
+      getNativeElementProps(as || 'div', {
+        ref,
+        ...props,
+      }),
+      { elementType: 'div' },
+    ),
+    icon: slot.optional(icon, { elementType: 'div' }),
```

Comment:
also i know this is probably out of scope of this PR since this api is already set :)

Reply:
Yeah, this API has been discussed for a while now, let me share some points here:

1. `elementType` was optional on an initial implementation, we ended up deciding to make it required as otherwise people would be forgetting to actually add the element type of a slot, as the `state.components[slot]` is to be deprecated
2. we don't have on our API no method inside of another method, so although this whole idea (`slot` and then `slot.something`) was discussed, for the sake of coherence we decided to not go that way.